app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb

```diff
@@ -40,11 +40,13 @@ def configuration_profiles
   # }]
   def configured_systems
     collector.virtual_machines.each do |virtual_machine|
+      virtual_instance_ref = virtual_machine["idFromProvider"].to_s
       persister.configured_systems.build(
         :manager_ref          => virtual_machine["id"].to_s,
         :name                 => virtual_machine["name"],
         :ipaddress            => virtual_machine["ipaddresses"]&.first,
-        :virtual_instance_ref => virtual_machine["idFromProvider"]
+        :virtual_instance_ref => virtual_instance_ref,
+        :counterpart          => persister.vms.lazy_find(virtual_instance_ref)
```

Comment:
If the `virtual_machine["idFromProvider"]` is `nil` then we're going to query for VMs with a `""` `:uid_ems` because of the `.to_s` which is probably not what we want.  At best it is a wasted query at worst it might actually hit on a record that isn't related to this at all.

I think we would be better off with:
```
virtual_instance_ref = virtual_machine["idFromProvider"]
counterpart          = persister.vms.lazy_find(virtual_instance_ref) if virtual_instance_ref

persister.configured_systems.build(
  ...
  :virtual_instance_ref => virtual_instance_ref,
  :counterpart          => counterpart
  ...
)
```

This way if the idFromProvider is nil we won't set either the instance_ref or the counterpart, wdyt?

Reply:
Thanks for the suggestions Adam. I incorporated both in the latest PR.